src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -51,8 +52,8 @@
   private final BazelBspServer bspServer;
   private final DiagnosticsDispatcher diagnosticsDispatcher;
 
-  private final Stack<TaskId> taskParkingLot = new Stack<>();
   private final Set<Uri> compilerClasspath = new TreeSet<>();
+  private final Deque<TaskId> startedEventTaskIds = new ArrayDeque<>();
```

Comment:
https://stackoverflow.com/questions/12524826/why-should-i-use-deque-over-stack

It can be any collection, actually it would be best not to have any collection here if it is allowed only to have one element but [I don't know how to do it](https://github.com/JetBrains/bazel-bsp/pull/21#discussion_r537079701).